GoogleForCreators/web-stories-wp
# Story Details Modal: Fix inconsistencies in Story Title

## Context

@swissspidy pointed out ahead of bug bash tomorrow 2 inconsistencies with the story details modal with the editor. The inconsistencies are from following figma and not properly discussing design's idea of making the Story Title required. 

If story titles are going to be actually required we will do that separately. These are things I should have caught early on and did not.

## Summary

Get us back into parity with the editor re story title.

## Relevant Technical Choices

Now that the story details modal publish button is always enabled we can move all of the story hooks into the component they're used in instead of having to pass props around, yay! Most of this is just clean up because of that. 

Also eliminating the char count cap on story title, since the editor doesn't show that. 

## To-do



## User-facing changes

| Before | After | 
| --- | --- | 
| <img width="689" alt="Screen Shot 2022-02-21 at 11 26 35 AM" src="https://user-images.githubusercontent.com/10720454/155009526-081de83a-8ef1-4b86-9bba-59e6d2adfc97.png"> | <img width="677" alt="Screen Shot 2022-02-21 at 11 27 49 AM" src="https://user-images.githubusercontent.com/10720454/155009675-12c8b772-66f1-4416-a1d4-dd5baac5c92d.png"> | 

## Testing Instructions

- Enable story details modal experiment 
- See that you can publish a story without a title
- See that you can publish a story with a very long title 
- permalink updates when permalink doesn't already exist and story title blur happens 

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10644 
Fixes #10647
